### PR TITLE
[prometheus] deckhouse web interfaces list

### PR DESCRIPTION
**Pull request by yalosev**
Signed-off-by: Yuriy Losev <yuriy.losev@flant.com>

## Description
Generate list of deckhouse public web pages (dashboard, docs, prometheus, istio/kiali, etc) and create a table with links on a grafana home page
image

## Why do we need it, and what problem does it solve?
It would be comfortable for a new users to see all resources they might need in one place

## Changelog entries

```changes
section: prometheus
type: feature
summary: Create table with enabled Deckhouse web interfaces on the Grafana home page
```

### CODE REVIEW

**Comment by nabokihms** [comment]
image
1. It looks like this new panel does not fit the screen. 
2. Do we need a header for the table? It seems clear where the name is and where the URL is to me.
3. Other tables have transparent backgrounds. It would be best to align this new table with others.

File: modules/300-prometheus/hooks/metrics_web_interfaces.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+	rule := ing.Spec.Rules[0]
+
+	host := rule.Host
+	urlPath := ""
+	if len(rule.HTTP.Paths) > 0 {
+		urlPath = rule.HTTP.Paths[0].Path
+	}
+
+	if urlPath != "" {
```
Comment (nabokihms) [inline modules/300-prometheus/hooks/metrics_web_interfaces.go:91]: Seems like a hack to me. Maybe it is better to pass the path explicitly somehow (via annotations?).

File: modules/500-upmeter/templates/status/ingress.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@ kind: Ingress
 metadata:
   name: status
   namespace: d8-{{ .Chart.Name }}
-  {{- include "helm_lib_module_labels" (list . (dict "app" "status")) | nindent 2 }}
+  {{- include "helm_lib_module_labels" (list . (dict "app" "status" "deckhouse.io/export-domain" "status")) | nindent 2 }}
```
Comment (nabokihms) [inline modules/500-upmeter/templates/status/ingress.yaml:8]: Let's do the same for the upmeter domain.

Comment (shvgn) [reply modules/500-upmeter/templates/status/ingress.yaml:8]: I expected it to be an annotation. Why is it a label?

Comment (yalosev) [reply modules/500-upmeter/templates/status/ingress.yaml]: never mind. It became an annotation :)

File: modules/300-prometheus/hooks/metrics_web_interfaces.go (hunk 1 above)
Comment (yalosev) [reply modules/300-prometheus/hooks/metrics_web_interfaces.go]: Added annotation. If annotation is not set - get it from rules

**Comment by yalosev** [comment]
image

File: ee/fe/modules/500-openvpn/templates/openvpn/ingress.yaml [hunk 3]
```diff
@@ -7,6 +7,8 @@ metadata:
   namespace: d8-{{ .Chart.Name }}
   {{- include "helm_lib_module_labels" (list . (dict "app" .Chart.Name)) | nindent 2 }}
   annotations:
+    web.deckhouse.io/export-name: "openvpn"
+    web.deckhouse.io/export-icon: "https://openvpn.net/wp-content/uploads/cropped-openvpn-32x32.png"
```
Comment (nabokihms) [inline ee/fe/modules/500-openvpn/templates/openvpn/ingress.yaml:11]: @konstantin-axenov, we need a second opinion here. We decided to add icons for all interfaces, and it looks marvelous.


However, we have a problem with icons providers.There are two options and corresponding perks:
1. We can use external links for icons. It is easy to add new interfaces, but we cannot control these icons. If someone decides to delete an icon, we will have to release a new version of Deckhouse to fix it. There is also a problem: users need access to websites with these icons (which will not work for closed environments). 

2. On the other hand, we can add all icons to the Grafana pod and serve them as static images by the Grafana server itself. This approach is solid, yet it makes adding a new interface a little more complicated.

WDYT?

Comment (konstantin-axenov) [reply ee/fe/modules/500-openvpn/templates/openvpn/ingress.yaml:11]: I think that the second option is the right way to do it.

Comment (konstantin-axenov) [reply ee/fe/modules/500-openvpn/templates/openvpn/ingress.yaml:11]: BTW, it looks really nice!

Comment (shvgn) [reply ee/fe/modules/500-openvpn/templates/openvpn/ingress.yaml:11]: Upmeter has its icon tho [modules/500-upmeter/images/webui/src/assets/upmeter.png](https://github.com/deckhouse/deckhouse/blob/main/modules/500-upmeter/images/webui/src/assets/upmeter.png)

Comment (yalosev) [reply ee/fe/modules/500-openvpn/templates/openvpn/ingress.yaml]: Done

**Review by nabokihms — APPROVED** [review]

**Comment by nabokihms** [comment]
I realized that anyone could add their web interfaces to the front page by annotation desired ingress resources. Good customization feature!

**Pull request merged**